src/services/formatting/formattingScanner.ts

```diff
@@ -195,30 +172,42 @@ namespace ts.formatting {
-                lastScanAction = ScanAction.RescanGreaterThanToken;
-            }
-            else if (expectedScanAction === ScanAction.RescanSlashToken && startsWithSlashToken(currentToken)) {
-                currentToken = scanner.reScanSlashToken();
-                Debug.assert(n.kind === currentToken);
-                lastScanAction = ScanAction.RescanSlashToken;
-            }
-            else if (expectedScanAction === ScanAction.RescanTemplateToken && currentToken === SyntaxKind.CloseBraceToken) {
-                currentToken = scanner.reScanTemplateToken();
-                lastScanAction = ScanAction.RescanTemplateToken;
-            }
-            else if (expectedScanAction === ScanAction.RescanJsxIdentifier && currentToken === SyntaxKind.Identifier) {
-                currentToken = scanner.scanJsxIdentifier();
-                lastScanAction = ScanAction.RescanJsxIdentifier;
-            }
-            else if (expectedScanAction === ScanAction.RescanJsxText) {
-                currentToken = scanner.reScanJsxToken();
-                lastScanAction = ScanAction.RescanJsxText;
-            }
-            else {
-                lastScanAction = ScanAction.Scan;
+            lastScanAction = ScanAction.Scan;
+            switch (expectedScanAction) {
+                case ScanAction.RescanGreaterThanToken:
+                    if (currentToken === SyntaxKind.GreaterThanToken) {
+                        currentToken = scanner.reScanGreaterToken();
+                        Debug.assert(n.kind === currentToken);
+                        lastScanAction = ScanAction.RescanGreaterThanToken;
+                    }
+                    break;
+                case ScanAction.RescanSlashToken:
+                    if (startsWithSlashToken(currentToken)) {
+                        currentToken = scanner.reScanSlashToken();
+                        Debug.assert(n.kind === currentToken);
+                        lastScanAction = ScanAction.RescanSlashToken;
+                    }
+                    break;
+                case ScanAction.RescanTemplateToken:
+                    if (currentToken === SyntaxKind.CloseBraceToken) {
+                        currentToken = scanner.reScanTemplateToken();
+                        lastScanAction = ScanAction.RescanTemplateToken;
+                    }
+                    break;
+                case ScanAction.RescanJsxIdentifier:
+                    if (currentToken === SyntaxKind.Identifier) {
+                        currentToken = scanner.scanJsxIdentifier();
+                        lastScanAction = ScanAction.RescanJsxIdentifier;
+                    }
+                    break;
+                case ScanAction.RescanJsxText:
+                    currentToken = scanner.reScanJsxToken();
+                    lastScanAction = ScanAction.RescanJsxText;
+                    break;
+                case ScanAction.Scan:
+                    break;
+                default:

```

Comment:
Can we remove line 175 and put it inside the default? The case ScanAction.Scan wont be necessary either with this.

Reply:
Some of the cases have `if`s that result in them not setting `lastScanAction`, so if we moved it to the `default` then `lastScanAction` would retain its old value. I thought it would be easier to set `lastScanAction` once before the `switch` than to give every `if` an `else { lastScanAction = ScanAction.Scan; }`.

Reply:
I've moved the whole switch statement to a function, which should be easier to read.

Reply:
I see what you mean and you're right. Thanks for clarifying.